src/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -313,7 +313,15 @@ private string GetDotnetExeDirectory()
             string environmentOverride = _getEnvironmentVariable("DOTNET_MSBUILD_SDK_RESOLVER_CLI_DIR");
             if (environmentOverride != null)
```

Comment:
Shouldn't this just be `(!string.IsNullOrEmpty(environmentOverride))`, and then we don't need the try/catch further down?

Reply:
There are other reasons `Path.GetDirectoryName` could fail, like if it has invalid characters.

Reply:
From the linked doc issue, it sounded like the problem was that GetEnvironmentVariable was returning an empty string instead of null.

Silently falling back to the installed SDK may not be what we want to do when there is actually an invalid value in this environment variable, perhaps due to a typo.

Reply:
In my case the environment return an empty string but how said @Forgind it's possible that the path contains invalid characters, and the try/cathc prevent any issu related with this variable.

Reply:
I think the goal of this change should be fail fast and return a more meaningful message to guide the user to set this env correctly or uncover the bug faster(since it is also set by the CLI itself).

Use `Path.GetDirectoryName(environmentOverride);` to trigger an exception is smart but I think it is more clear to just use Directory.Exists() and throw with a better error message when it is not.

I don't have a strong opinion about check null vs null and empty. We don't ignore empty consistently.